test/test_selectors.py

```diff
@@ -0,0 +1,672 @@
+        self.assertEqual(key.events, key2.events)
+        self.assertIsNot(key.data, key2.data)
+        self.assertEqual(key2, s.get_key(rd))
+        self.assertIs(key2.data, d2)
+
+        # Modify invalid fileobj
+        self.assertRaises(KeyError, s.modify, 999999, selectors.EVENT_READ)
+
+    def test_empty_select(self):
+        s = self.make_selector()
+        self.assertEqual([], s.select(timeout=SHORT_SELECT))
+
+    def test_select_multiple_event_types(self):
+        s = self.make_selector()
+
+        rd, wr = self.make_socketpair()
+        key = s.register(rd, selectors.EVENT_READ | selectors.EVENT_WRITE)
+
+        self.assertEqual([(key, selectors.EVENT_WRITE)], s.select(0.001))
+
+        wr.send(b'x')
+        time.sleep(0.01)  # Wait for the write to flush.
+
+        self.assertEqual([(key, selectors.EVENT_READ | selectors.EVENT_WRITE)], s.select(0.001))
+
+    def test_select_multiple_selectors(self):
+        s1 = self.make_selector()
+        s2 = self.make_selector()
+        rd, wr = self.make_socketpair()
+        key1 = s1.register(rd, selectors.EVENT_READ)
+        key2 = s2.register(rd, selectors.EVENT_READ)
+
+        wr.send(b'x')
+        time.sleep(0.01)  # Wait for the write to flush.
+
+        self.assertEqual([(key1, selectors.EVENT_READ)], s1.select(timeout=0.001))
+        self.assertEqual([(key2, selectors.EVENT_READ)], s2.select(timeout=0.001))
+
+    def test_select_no_event_types(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        self.assertRaises(ValueError, s.register, rd, 0)
+
+    def test_select_many_events(self):
+        s = self.make_selector()
+        readers = []
+        writers = []
+        for i in range(32):
+            rd, wr = self.make_socketpair()
+            readers.append(rd)
+            writers.append(wr)
+            s.register(rd, selectors.EVENT_READ)
+
+        self.assertEqual(0, len(s.select(0.001)))
+
+        # Write a byte to each end.
+        for wr in writers:
+            wr.send(b'x')
+
+        # Give time to flush the writes.
+        time.sleep(0.01)
+
+        ready = s.select(0.001)
+        self.assertEqual(32, len(ready))
+        for key, events in ready:
+            self.assertEqual(selectors.EVENT_READ, events)
+            self.assertIn(key.fileobj, readers)
+
+        # Now read the byte from each endpoint.
+        for rd in readers:
+            data = rd.recv(1)
+            self.assertEqual(b'x', data)
+
+        self.assertEqual(0, len(s.select(0.001)))
+
+    def test_select_timeout_none(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(wr, selectors.EVENT_WRITE)
+
+        with self.assertTakesTime(upper=SHORT_SELECT):
+            self.assertEqual(1, len(s.select(timeout=None)))
+            self.assertEqual(1, len(s.select(timeout=None)))
+            self.assertEqual(1, len(s.select(timeout=None)))
```

Comment:
Why are there three calls here? Might be good to add a comment if you think they're necessary.